### PR TITLE
GL: fix non-default indexed colormaps

### PR DESCRIPTION
**Pull request by bkoropoff**
Colormap lookups are done in the fragment shader, so don't do them when generating the texture.

Closes issue #207

### CODE REVIEW

**Comment by kraflab** [comment]
This fixes 207 but it breaks the invuln colormap unfortunately

**Comment by bkoropoff** [comment]
Not sure why invuln is handled differently, but I added a check for it.  Also expanded the fix to raws and patches.

File: prboom2/src/gl_texture.c [hunk 1]
```diff
@@ -433,40 +436,26 @@ static void gld_AddPatchToTexture_UnTranslated(GLTexture *gltexture, unsigned ch
...
-            }
-            else
-            {
-              buffer[pos+0]=playpal[colormap[source[j]]*3+0];
-              buffer[pos+1]=playpal[colormap[source[j]]*3+1];
-              buffer[pos+2]=playpal[colormap[source[j]]*3+2];
-            }
+            // [XA] new indexed color mode: store the palette index in
+            // the R channel and get the colormap index in the shader.
+            buffer[pos + 0] = gltexture->player_cm == INVULN_PLAYER_CM ? colormap[source[j]] : source[j];
```
Comment (kraflab) [inline prboom2/src/gl_texture.c:443]: I was wondering why we don't need a comp_sky check here, but there is a separate gld_AddIndexedSkyToTexture covering indexed mode. Just mentioning this for posterity.

**Comment by kraflab** [comment]
Looks great 👍 
As far as I know, this was the last remaining error in indexed mode, it should now have parity 🥳

**Pull request merged**